### PR TITLE
[all] Externalize metis

### PR DESCRIPTION
**Pull request by olivier-roussel**
Tackles one point of #4221. Required for the conda-forge package (https://github.com/conda-forge/staged-recipes/pull/23085).
Externalize the previously embedded version of metis (5.1.0) which was statically linked.

Provides a cmake module finder file (FindMetis.cmake), tested successfully with the following ways to provide the library:
- compile & install from source (v5.1.1) (warning: official repository cmakefiles are broken)
- APT package libmetis-dev (v5.1.0)
- conda package (v5.1.0 and v5.1.1)
- vcpkg package (v2022-07-27)
All on ubuntu 22.04. Not tested with conan.

Some remarks to embed metis in the CI: The embedded version was compiled with GKlib using the `GKRAND` option. To the best of my knowledge, this can be provided either from compiling from source or with the conan package with a specific option.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: Sofa/Component/LinearSolver/Direct/CMakeLists.txt [hunk 1]
```diff
@@ -59,12 +59,7 @@ set(SOURCE_FILES
 add_subdirectory(extlibs/csparse)
 sofa_set_01(SOFA_COMPONENT_LINEARSOLVER_DIRECT_HAVE_CSPARSE VALUE TRUE)
 
-sofa_find_package(metis QUIET) # Unix users can have an installed version of metis
-if(NOT metis_FOUND)
-    message(STATUS "${PROJECT_NAME}: using built-in metis library")
-    option(METIS-GKLIB_GKRAND "enable GKRAND support" ON) #GKRAND support allows the portability of random number generation across different architectures
-    add_subdirectory(extlibs/metis-5.1.0)
-endif()
+sofa_find_package(Metis REQUIRED) # Unix users can have an installed version of metis
```
Comment (alxbilger) [inline Sofa/Component/LinearSolver/Direct/CMakeLists.txt]:
```suggestion
sofa_find_package(Metis REQUIRED)
```

**Review by alxbilger — COMMENTED** [review]
`GKRAND` is rather important if we want to keep the same simulation results across different architectures.

**Comment by olivier-roussel** [comment]
> `GKRAND` is rather important if we want to keep the same simulation results across different architectures.

OK. So we can still build our binaries with it by compiling metis from source on the CI and setting `GKRAND` instead of using distribution package. However, we will have to notify conda users that `GKRAND` will not be enabled with the SOFA conda package.

**Comment by olivier-roussel** [comment]
Closed as replaced by #4450

**Pull request closed without merging**